src/lib.rs

```diff
@@ -645,6 +686,37 @@ impl SystemHandle {
         }
     }
 
+    /// Subscribe given `recipient` to events of type `E`. See [`Context::subscribe()`].
+    pub fn subscribe_recipient<M: 'static, E: Event + Into<M>>(&self, recipient: Recipient<M>) {
+        let mut event_subscribers = self.event_subscribers.write();
+
+        let subs = event_subscribers.events.entry(TypeId::of::<E>()).or_default();
+
+        subs.push(Box::new(move |e| {
+            if let Some(event) = e.downcast_ref::<E>() {
+                let msg = event.clone();
+                recipient.send(msg.into())?;
+            }
+
+            Ok(())
+        }));
+    }
+
+    /// Publish an event. All actors that have previously subscribed to the type will receive it.
+    ///
+    /// When sending to some subscriber fails, this method returns an error immediately.
```

Comment:
Is this always desirable? I imagine you'll likely always want to panic if this happens, but I'm not sure if I like that this will deliver the publish to all receivers before the erroring one but to none after it. If someone tries to recover gracefully from this error, they might have a hard time knowing the state of the system precisely.